hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -210,6 +232,11 @@ protected String tableName() {
     return fullName;
   }
 
+  @Override
+  protected EncryptionManagerFactory encryptionManagerFactory() {
```

Comment:
Why is this exposing the factory? Shouldn't this use the factory to return an encryption manager [here](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableOperations.java#L72)?

Reply:
@rdblue, very good spot! During my first iteration, I was putting this method in `TableOperations` and use the factory to return the manager, and this will make `TableOperations` have two encryption related methods, some of my colleagues are worrying about breaking change to `TableOperations` as public API. Myself I vote also to move this method to `TableOperations`  cc @stevenzwu @ggershinsky

Reply:
@rdblue , calling encryptionManagerFactory.create() will be a duplicated logic for table operations sub classes. That's why I put the duplicated logic to `BaseMetastoreTableOperations` class and in that class I defined method `protected abstract EncryptionManagerFactory encryptionManagerFactory()`. I would think the factory create logic to `TableOperations` directly and that would require to make `TableOperations` as an abstract class, need your advice on this.